redpanda-otel-exporter/exporter.go

```diff
@@ -0,0 +1,294 @@
+package redpandaotelexporter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"google.golang.org/protobuf/proto"
+)
+
+// SerializationFormat specifies the format to use when serializing telemetry data
+type SerializationFormat int
+
+const (
+	// SerializationFormatJSON uses JSON encoding (default)
+	SerializationFormatJSON SerializationFormat = iota
+	// SerializationFormatProtobuf uses Protocol Buffers encoding (OTLP format)
+	SerializationFormatProtobuf
+)
+
+// String returns the string representation of the serialization format
+func (f SerializationFormat) String() string {
+	switch f {
+	case SerializationFormatJSON:
+		return "json"
+	case SerializationFormatProtobuf:
+		return "protobuf"
+	default:
+		return "unknown"
+	}
+}
+
+// config holds the configuration for the Kafka exporter.
+type config struct {
+	brokers             []string
+	topic               string
+	clientID            string
+	timeout             time.Duration
+	resource            *resource.Resource
+	serializationFormat SerializationFormat
+	kafkaOptions        []kgo.Opt
+}
+
+// Option is a function that configures an Exporter
+type Option func(*config)
+
+// WithBrokers sets the Kafka broker addresses (required)
+func WithBrokers(brokers ...string) Option {
+	return func(c *config) {
+		c.brokers = brokers
+	}
+}
+
+// WithClientID sets the Kafka client ID (default: "otel-kafka-exporter")
+func WithClientID(clientID string) Option {
+	return func(c *config) {
+		c.clientID = clientID
+	}
+}
+
+// WithTimeout sets the timeout for export operations (default: 30s)
+func WithTimeout(timeout time.Duration) Option {
+	return func(c *config) {
+		c.timeout = timeout
+	}
+}
+
+// WithResource sets the OpenTelemetry resource
+func WithResource(resource *resource.Resource) Option {
+	return func(c *config) {
+		c.resource = resource
+	}
+}
+
+// WithSerializationFormat sets the serialization format (JSON or Protobuf)
+func WithSerializationFormat(format SerializationFormat) Option {
+	return func(c *config) {
+		c.serializationFormat = format
+	}
+}
+
+// WithKafkaOptions sets additional franz-go client options
+func WithKafkaOptions(opts ...kgo.Opt) Option {
+	return func(c *config) {
+		c.kafkaOptions = opts
+	}
+}
+
+// WithTopic sets the Kafka topic name (overrides the default)
+func WithTopic(topic string) Option {
```

Comment:
Order of functions is preserved but for this one. I wonder if in general it was not just easier to export the full config?